shared/common-adapters/icon.native.js

```diff
@@ -91,6 +89,28 @@ const Image = glamorous.image(
     props.style && props.style.backgroundColor ? {backgroundColor: props.style.backgroundColor} : null
 )
 
+const stripStyles = (styleArg: CollapsibleStyle, toStrip) => {
```

Comment:
we shouldn't be doing this at all. this'll not work in our modern styling. on mobile the style will get turned into an id so you can't introspect it and we shouldn't be doing it  so color has to be its own prop like hovercolor

Reply:
What about all the other props we pass down to `ClickableBox` below?

Reply:
yeah that was going to be my question too

Reply:
What i was mentioning during planning around this ticket was:

When we have a component that naturally has a container and some inner stuff (like this) if we're going to expose a style override the style override itself needs to be split. We do this in other places. 
This pattern of one style object which we then pull things out of never worked well and it for sure won't when we use StyleSheet create on mobile.

So really this class needs style and containerStyle or something

Reply:
Part of the difficulty is that there's a lot of conditionals here that show different things on different conditions, e.g. the ClickableBox is only there when there's a click handler. And the icon can be either text or an image. If we split up the style object, we'll need like 3 or 4 different ones, and it'll be a pretty leaky abstraction. So I'm not exactly sure what the best way to do this is...

Reply:
It seems to me that some of what you describe isn't behavior that we want and does cause implementation headaches and, i'm sure, user confusion.

in my mind this is just a component and it has an image inside a container (if its an icon and its text thats an implementation detail no one externally cares about). You likely want to affect its style in terms of layout and in terms of affecting the image. 

Given that I'd advocate there's 1 style prop that is applied to the container (more layout stuff) and things that affect the image itself (color, fontsize, etc) would be props.

I think having a clickable box or not should depend on the onClick existing, but there should likely be a box no matter what

Reply:
I was in the process of moving color, fontSize, and textAlign up to be peers of style, e.g.:
```
-          <Icon style={_createIcon} type="iconfont-new" onClick={props.onCreate} />
+          <Icon style={_createIcon} type="iconfont-new" onClick={props.onCreate} color={globalColors.blue} />
```
but it sounds like we want a containerStyle object instead in all those cases? like
```
<Icon style={_createIcon} type="iconfont-new" onClick={props.onCreate} containerStyle={{color: globalColors.blue}} />
```
? I'm still getting up to speed on the pros and cons

Reply:
color / fontsize affect the icon itself so they'd be props. i think text align could likely be ditched and replaced with some kind of layout styling (like alignItems)
i'm saying style would be the containerstyle